qiskit/primitives/base/base_estimator.py

```diff
@@ -254,3 +328,48 @@ def parameters(self) -> tuple[ParameterView, ...]:
             Parameters, where ``parameters[i][j]`` is the j-th parameter of the i-th circuit.
         """
         return tuple(self._parameters)
+
+
+BaseEstimator = BaseEstimatorV1
+
+
+class BaseEstimatorV2:
+    """Estimator base class version 2.
+
+    An estimator estimates expectation values for provided quantum circuit and
+    observable combinations.
+    """
+
+    def __init__(self, precision: float):
```

Comment:
I understand the intent to include precision. But, why did you not let PUB have this? In other words, given multiple PUBs, wouldn't you want to change the precision for each PUB?
Also, is it not optional? Do you ever want to set the accuracy on the shots side?

Reply:
In the latest round, this arg has been removed from the constructor and added to the run method and to the pub attributes. Ihe value in the pub, if present, takes precedence. The value in the run kwargs, if present sets the value for each pub where it is not set. The primitive must choose a value for precision if none are given, and this is left up to each implementation to decide. For example, state vector simulators have no reason to have precision >0 (except, of course, machine epsilon, which we can ignore).